File: 2-second-year-2022-2023/Programming-languages-and-laboratory/Java/Exercises/37_compito_java_2020_01_24/Tennis.java
public class Tennis extends Campo {
	private float temperaturaMedia;
	private String terreno;
	private float costoOrario;

	public Tennis(int codice, String sport, String nome, float larghezza, float lunghezza, float temperaturaMedia, String terreno, float costoOrario) {
		super(codice, sport, nome, larghezza, lunghezza, costoOrario);
		this.temperaturaMedia = temperaturaMedia;
		this.terreno = terreno;
	}

	public String toString() {
		return sport + "\t" + nome + "\t" + codice + "\t" + larghezza + "\t" + lunghezza + "\t" + temperaturaMedia
				+ "\t" + terreno + "\t-\t-\t" + costoOrario;
	}
}

File: 2-second-year-2022-2023/Programming-languages-and-laboratory/Python/Lessons/26_20221202/37_GestioneCircoloSportivo/Tennis.py
from Campo import Campo


class Tennis(Campo):
    def __init__(self, codice, sport, nome, larghezza, lunghezza, temperatura, terreno, costo):
        super().__init__(codice, sport, nome, larghezza, lunghezza)
        self._temperatura = temperatura
        self._terreno = terreno
        self._costo = costo
        self._numero_prenotazioni = 0

    def incrementa_prenotazione(self):
        self._numero_prenotazioni += 1

    def get_incasso(self):
        return self._costo * self._numero_prenotazioni

    def __str__(self):
        return super().__str__() + "\t" + str(self._temperatura) + "\t" + self._terreno + "\t-\t-\t" + str(self._costo)
